recipes/cpr/all/conanfile.py

```diff
@@ -0,0 +1,78 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class CprConan(ConanFile):
+    name = "cpr"
+    description = "C++ Requests: Curl for People, a spiritual port of Python Requests"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://whoshuu.github.io/cpr/"
+    license = "MIT"
+    topics = ("conan", "cpr", "requests", "web", "curl")
+    exports_sources = ["CMakeLists.txt", "patches/*"]
+    generators = "cmake", "cmake_find_package"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_openssl": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_openssl": True,
+    }
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        if self.options.with_openssl:
+            # If using OpenSSL, we need it to be active in libcurl too
+            if self.options["libcurl"].with_openssl == False:  # 'is False' doesn't work here (see conan-io/conan#3620)
+                raise ConanInvalidConfiguration("libcurl must be built with openssl support")
+            self.options["libcurl"].with_openssl = True
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        os.rename("cpr-{}".format(self.version), self._source_subfolder)
+
+    def requirements(self):
+        self.requires("libcurl/7.67.0")
+        if self.options.with_openssl:
+            self.requires("openssl/1.1.1d")
+
+    def _patch_sources(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["USE_SYSTEM_CURL"] = True
```

Comment:
use system curl? Do you know why this is set to true?

Reply:
IIRC it has an embedded version.

Reply:
Yes, it is needed so it uses `find_package`, otherwise it uses a submodule/vendorized version.